sympy/geometry/line.py

```diff
@@ -1070,6 +1070,39 @@ def contains(self, other):
             return Point.is_collinear(self.p1, self.p2, other.p1, other.p2)
         return False
 
+    def coordinate(self, other):
```

Comment:
Would it make sense to make the arg list be `self, x, y=None` and then `line.coordinate(3)` would give `t` where the Point on the line whose x value is 3 would be returned? (It would raise an error if the line was vertical and didn't go through `x=3`. `line.coordinate(3, 4)` would behave as it does now. And `line.coordinate(y=4)` would behave the same as the `x=` case but raise an error for a horizontal line not passing through 4.

Reply:
If I understand you correctly that would involve essentially two steps: 1) from plane coordinate x to point P on the line, 2) from P to its affine line coordinate. Perhaps there could be another function for step 1) alone?

Reply:
Yes, I see your point. I didn't appreciate the difference in dealing with 2D vs 3D lines. It need not be part of this PR. I'll leave the final commit to you on this one.